src/sentry/search/events/builder.py

```diff
@@ -1645,12 +1660,133 @@ def __init__(
 
 
 class SessionsQueryBuilder(QueryBuilder):
+    # ToDo(ahmed): Rename this to AlertsSessionsQueryBuilder as it is exclusively used for crash rate alerts
     def resolve_params(self) -> List[WhereType]:
         conditions = super().resolve_params()
         conditions.append(Condition(self.column("org_id"), Op.EQ, self.organization_id))
         return conditions
 
 
+class SessionsV2QueryBuilder(QueryBuilder):
+    condition_fields = {"project", "project_id", "environment", "release"}
+
+    def __init__(
+        self,
+        *args: Any,
+        granularity: Optional[int] = None,
+        extra_condition_fields: Optional[Sequence[str]] = None,
+        **kwargs: Any,
+    ):
+        self._granularity = granularity
+        self._extra_condition_fields = extra_condition_fields or []
+        self._non_nullable_keys = self.condition_fields
+        # Fields that if present in the select, then need to be in the group by as well
+        self.required_select_groupby = copy.copy(self.condition_fields)
+        self.required_select_groupby.remove("project")
+        super().__init__(*args, **kwargs)
+
+    def resolve_params(self) -> List[WhereType]:
+        conditions = super().resolve_params()
+        conditions.append(Condition(self.column("org_id"), Op.EQ, self.organization_id))
+        return conditions
+
+    def resolve_groupby(self, groupby_columns: Optional[List[str]] = None) -> List[SelectType]:
+        if groupby_columns is None:
+            groupby_columns = []
+        return list(
+            {col for col in self.columns if col.name in self.required_select_groupby}
+            | {self.resolve_column(column) for column in groupby_columns}
+        )
+
+    def resolve_granularity(self) -> Optional[Granularity]:
+        if self._granularity is None:
+            return None
+        return Granularity(self._granularity)
+
+    def resolve_query(
+        self,
+        query: Optional[str] = None,
+        use_aggregate_conditions: bool = False,
+        selected_columns: Optional[List[str]] = None,
+        groupby_columns: Optional[List[str]] = None,
+        equations: Optional[List[str]] = None,
+        orderby: Optional[List[str]] = None,
+    ) -> None:
+        super().resolve_query(
+            query, use_aggregate_conditions, selected_columns, groupby_columns, equations, orderby
+        )
+        with sentry_sdk.start_span(op="QueryBuilder", description="resolve_granularity"):
+            # Needs to happen before params and after time conditions since granularity can change start&end
+            self.granularity = self.resolve_granularity()
+
```

Comment:
With the super.resolve_query on 1715, isn't this happening twice now?

Also I think this comment is incorrect now, since granularity is being resolved after start&end